api/users/usersController.js

```diff
@@ -42,5 +43,46 @@ module.exports.factory = (
       })
   })
 
+  router.put('/users/:email/addcat', function (req, res) {
```

Comment:
In REST URIs are a nice representation of a resource. Your current format isn't RESTful as it includes 'addcat' and 'remcat' The add category should be a PUT as you listed. I would suggest the URI being '/users/category/:email' This is now saying update the category resource at email (unique ID).